File: examples/erlang/user-server/test/user_server_probe.erl
-module(user_server_probe).

-include("user_types.hrl").

-export([handle_get_user/3]).

%% Test double: echoes decoded input user id into the output user map.
handle_get_user(_Ctx, #get_user_input{user_id = UserId}, _Meta) ->
    {ok, #get_user_output{
        user = #{<<"userId">> => UserId}
    }}.

File: examples/elixir/user-server/test/support/user_server_probe.ex
defmodule UserServerProbe do
  @moduledoc false

  @spec handle_get_user(term(), User.GetUserInput.t(), term()) ::
          {:ok, User.GetUserOutput.t()} | {:error, term()}
  def handle_get_user(_ctx, input, _meta) do
    {:ok, %User.GetUserOutput{user: %{"userId" => input.user_id}}}
  end
end
